File: src/main/java/com/thebluehats/server/core/modules/BowManagerModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.game.managers.combat.BowManager;

public class BowManagerModule extends AbstractModule {
    @Provides
    @Singleton
    static BowManager provideBowManager() {
        return new BowManager();
    }

    @Override
    protected void configure() {
    }
}

File: minecraft-server/src/main/kotlin/com/thebluehats/server/core/modules/BowManagerModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.game.managers.combat.BowManager

class BowManagerModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        @JvmStatic
        fun provideBowManager(): BowManager {
            return BowManager()
        }
    }
}